fireapp/lib/presentation/unavailability_form/unavailability_form_widget.dart

```diff
@@ -67,125 +48,142 @@ class _UnavailabilityFormState extends FireAppState<UnavailabilityForm>
+                        builder: (context, data) {
+                          final isFormValid = isFormValidSnapshot.data ?? false;
+                          final submitButtonLabel = !isEditMode
+                              ? (AppLocalizations.of(context)
                                       ?.addUnavailabilityButton ??
-                                  "Add Schedule"),
-                              if (data is LoadingRequestState)
-                                SizedBox(
-                                  height: 8,
-                                  width: 8,
-                                  child: CircularProgressIndicator(
-                                      color: (Theme.of(context)
-                                              .primaryTextTheme
-                                              .titleLarge
-                                              ?.color ??
-                                          Colors.white)
-                                  ),
-                                )
-                            ],
-                          ),
-                        );
-                      },
-                    );
-                  })
-          )
-        ],
-        children: [
-          Form(
-            key: viewModel.formKey,
-            child: Column(
-              crossAxisAlignment: CrossAxisAlignment.start,
+                                  "Add Schedule")
+                              : (AppLocalizations.of(context)
+                                      ?.editUnavailabilityButton ??
+                                  "Add Edit");
+                          return ElevatedButton(
+                            style: ElevatedButton.styleFrom(
+                              minimumSize: const Size.fromHeight(40),
+                            ),
+                            onPressed:
+                                isFormValid ? viewModel.submitForm : null,
+                            child: Row(
+                              mainAxisAlignment: MainAxisAlignment.center,
+                              children: [
+                                Text(submitButtonLabel),
+                                if (data is LoadingRequestState)
+                                  SizedBox(
+                                    height: 8,
+                                    width: 8,
+                                    child: CircularProgressIndicator(
+                                        color: (Theme.of(context)
+                                                .primaryTextTheme
+                                                .titleLarge
+                                                ?.color ??
+                                            Colors.white)
+                                    ),
+                                  )
+                              ],
+                            ),
+                          );
+                        },
+                      );
+                    })
```

Comment:
The indentation for this code is strange and makes it a little hard to read, please fix